src/pages/Browse/index.spec.tsx

```diff
@@ -542,4 +548,157 @@ describe(__filename, () => {
   });
+
+  describe('preloading', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const setUpFilesAndRender = ({
+      beginFetchNextFile = false,
+      buildTree = true,
+      loadNextFile = false,
+      loadVersionFile = true,
+    }: {
+      beginFetchNextFile?: boolean;
+      buildTree?: boolean;
+      loadNextFile?: boolean;
+      loadVersionFile?: boolean;
+    } = {}) => {
+      const extraFileEntries = {
+        [file1]: { ...fakeVersionEntry, path: file1 },
+        [file2]: { ...fakeVersionEntry, path: file2 },
+      };
+      const {
+        _fetchVersionFile,
+        addonId,
+        fakeThunk,
+        store,
+        version,
+      } = setUpVersionFileUpdate({
+        extraFileEntries,
+        initialPath: file1,
+        loadVersionFile,
+      });
+      if (buildTree) {
+        store.dispatch(
+          fileTreeActions.buildTree({
+            version: createInternalVersion(version),
+          }),
+        );
+      }
+      if (beginFetchNextFile) {
+        store.dispatch(
+          versionsActions.beginFetchVersionFile({
+            path: file2,
+            versionId: version.id,
+          }),
+        );
+      }
+      if (loadNextFile) {
+        store.dispatch(
+          versionsActions.loadVersionFile({ path: file2, version }),
+        );
+      }
+      const dispatchSpy = spyOn(store, 'dispatch');
+
+      render({
+        _fetchVersionFile,
+        addonId: String(addonId),
+        store,
+        versionId: String(version.id),
+      });
+
+      return {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        store,
+        version,
+      };
+    };
+
+    it('dispatches fetchVersionFile for the next file', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        fakeThunk,
+        version,
+        dispatchSpy,
+      } = setUpFilesAndRender();
+
+      expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).toHaveBeenCalledWith({
+        addonId,
+        path: file2,
+        versionId: version.id,
+      });
+    });
+
+    it('does not dispatch fetchVersionFile for the next file if the current version file is not loaded', () => {
+      const { _fetchVersionFile, addonId, version } = setUpFilesAndRender({
+        loadVersionFile: false,
+      });
+
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+        addonId,
+        path: file2,
+        versionId: version.id,
+      });
+    });
+
+    it('does not dispatch fetchVersionFile for the next file when fileTree is not built', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        version,
+      } = setUpFilesAndRender({
+        buildTree: false,
+      });
+
+      expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+        addonId,
+        path: file2,
```

Comment:
Since the only important part here is that it's not called with the next path, you could simplify this test to:

```diff
diff --git a/src/pages/Browse/index.spec.tsx b/src/pages/Browse/index.spec.tsx
index 414e1ac..c60ee38 100644
--- a/src/pages/Browse/index.spec.tsx
+++ b/src/pages/Browse/index.spec.tsx
@@ -658,11 +658,9 @@ describe(__filename, () => {
       });
 
       expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
-      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
-        addonId,
-        path: file2,
-        versionId: version.id,
-      });
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith(
+        expect.objectContaining({ path: file2 }),
+      );
     });
 
     it('does not dispatch fetchVersionFile for the next file when the next file is loading', () => {

```

This applies to a couple other tests but I would keep at least one or two that do full assertions on the `versionId` and `addonId` parameters because it's important that `fetchVersionFile()` receives those when pre-loading.